src/main/java/seedu/address/logic/parser/RelateCommandParser.java

```diff
@@ -0,0 +1,51 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.model.person.Id.isValidId;
+
+import java.util.Arrays;
+
+import seedu.address.logic.commands.RelateCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.IdContainsDigitsPredicate;
+
+/**
+ * Parses input arguments and creates a new FindNumCommand object
+ */
+public class RelateCommandParser implements Parser<RelateCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the RelateCommand
+     * and returns a RelateCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public RelateCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+        if (trimmedArgs.isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, RelateCommand.MESSAGE_USAGE));
+        }
+
+        String[] providedIDs = trimmedArgs.split("\\s+");
+
+        // Check if all IDs in input are valid
+        for (String iDInList : providedIDs) {
+            if (!isValidId(Integer.parseInt(iDInList))) {
+                throw new ParseException(
+                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, RelateCommand.MESSAGE_USAGE));
```

Comment:
Can consider using `Id.MESSAGE_CONSTRAINTS` instead if the given number is not a valid `Id`